src/Workspaces/Core/Portable/Workspace/Solution/Document.cs

```diff
@@ -146,18 +157,22 @@ public bool SupportsSemanticModel
         /// <summary>
         /// Gets the <see cref="SyntaxTree" /> for this document asynchronously.
         /// </summary>
-        public Task<SyntaxTree> GetSyntaxTreeAsync(CancellationToken cancellationToken = default)
+        /// <returns>
+        /// The returned syntax tree can be null if the <see cref="SupportsSyntaxTree"/> returns false.</returns>
+        public Task<SyntaxTree?> GetSyntaxTreeAsync(CancellationToken cancellationToken = default)
         {
             // If the language doesn't support getting syntax trees for a document, then bail out immediately.
             if (!this.SupportsSyntaxTree)
             {
-                return SpecializedTasks.Default<SyntaxTree>();
+                return SpecializedTasks.Default<SyntaxTree?>();
             }
 
             // if we have a cached result task use it
             if (_syntaxTreeResultTask != null)
             {
-                return _syntaxTreeResultTask;
+                // _syntaxTreeResultTask is a Task<SyntaxTree> so the ! operator here isn't suppressing a possible null ref, but rather allowing the
+                // conversion from Task<SyntaxTree> to Task<SyntaxTree?> since Task itself isn't properly variant.
```

Comment:
💡 We should put thought into a _short_ comment covering this scenario, as I suspect it will not be uncommon.